### ISSUE TITLE
How to config the Timeout

### ISSUE THREAD

**Issue by ghost** [square/retrofit#641]
Hi, I have a problem on retrofit about to config the timeout with retrofit. My code like this:
        sBaseUrl = Configs.getString(Configs.BASE_URL);

```
    RestAdapter.Builder builder = new RestAdapter.Builder()
            .setEndpoint(sBaseUrl)
            .setConverter(new JacksonConverter())
            .setRequestInterceptor(
                    new ApiAuthInterceptor(MyApplication.getInstance()))
            .setErrorHandler(new ApiErrorHandler());

    builder.setLogLevel(LogLevel.FULL).setLog(new Log() {
        @Override
        public void log(String arg) {
            android.util.Log.d("rest", arg);
        }
    });
```

how to do it?? Is there a way to config the timeout no matter which kind of way to send a request such as okhttp or other??

**Comment by rpalsaxena** [comment]
I am getting timeout error  in logcat when i used retofit on my code... Could you please suggest me a solution.

**Comment by cassianomenezes** [comment]
Hey, @JakeWharton! I'm facing some timeout issues in my application.
I followed the documentation and examples online. I use Retrofit to make this REST call: 
@GET("/resolver/resolve")
    public ResolverResult resolve(@Query("id") String id);

My OkClient object is set like this:

OkHttpClient client = new OkHttpClient();
        client.setConnectTimeout(CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
        client.setReadTimeout(READ_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
        ArrayList<Protocol> protocols = new ArrayList<Protocol>();
        protocols.add(Protocol.HTTP_1_1);
        client.setProtocols(protocols);
        client.setFollowSslRedirects(true);
        client.setFollowRedirects(true);

In my Builder I'm setting the client as this okClient I just described (it works fine).

My idea is to catch timeout situations/exceptions when they happen. I want to show my user the timeout happened and he/she could try again later. 
To achieve this, I'm simulating very bad internet connection cases but Retrofit doesn't seem to follow the timeout values I've set. It just keeps trying again and again.

Am I missing something? Is there any other documentation I can follow to fix this?

Thank you.
